src/chem/aeronet_aod2ioda.py

```diff
@@ -0,0 +1,381 @@
+        else:
+            lat1 = str(float(self.latlonbox[0]))
+            lon1 = str(float(self.latlonbox[1]))
+            lat2 = str(float(self.latlonbox[2]))
+            lon2 = str(float(self.latlonbox[3]))
+            latlonbox = '&lat1=' + lat1 + '&lat2=' + \
+                lat2 + '&lon1=' + lon1 + '&lon2=' + lon2
+        print(base_url)
+        print(date_portion)
+        print(product)
+        print(inv_type)
+        print(time)
+        print(latlonbox)
+        if inv_type is None:
+            inv_type = ''
+        self.url = base_url + date_portion + product + \
+            inv_type + time + latlonbox + '&if_no_html=1'
+
+    def read_aeronet(self):
+        print('Reading Aeronet Data...')
+        df = pd.read_csv(self.url,
+                         engine='python',
+                         header=None,
+                         skiprows=6,
+                         parse_dates={'time': [1, 2]},
+                         date_parser=dateparse,
+                         na_values=-999)
+        columns = self.get_columns()
+        df.columns = columns
+        df.index = df.time
+        df.rename(columns={
+            'site_latitude(degrees)': 'latitude',
+            'site_longitude(degrees)': 'longitude',
+            'site_elevation(m)': 'elevation',
+            'aeronet_site': 'siteid'
+        },
+            inplace=True)
+        df.dropna(subset=['latitude', 'longitude'], inplace=True)
+        df.dropna(axis=1, how='all', inplace=True)
+        self.df = df
+
+    def get_columns(self):
+        header = pd.read_csv(self.url, skiprows=5, header=None,
+                             nrows=1).values.flatten()
+        final = ['time']
+        for i in header:
+            if "Date(" in i or 'Time(' in i:
+                pass
+            else:
+                final.append(i.lower())
+        return final
+
+    def add_data(self,
+                 dates=None,
+                 product='AOD15',
+                 latlonbox=None,
+                 daily=False,
+                 interp_to_aod_values=None,
+                 inv_type=None,
+                 freq=None,
+                 siteid=None):
+        self.latlonbox = latlonbox
+        self.siteid = siteid
+        if dates is None:  # get the current day
+            self.dates = pd.date_range(start=pd.to_datetime('today'),
+                                       end=pd.to_datetime('now'),
+                                       freq='H')
+        else:
+            self.dates = dates
+        self.prod = product.upper()
+        if daily:
+            self.daily = 20  # daily data
+        else:
+            self.daily = 10  # all points
+        if inv_type is not None:
+            self.inv_type = 'ALM15'
+        else:
+            self.inv_type = inv_type
+        if 'AOD' in self.prod:
+            self.new_aod_values = interp_to_aod_values
+        self.build_url()
+        try:
+            self.read_aeronet()
+        except Exception:
```

Comment:
if you can't read, just print the URL? Seems like it needs something else here.

Reply:
I have remove codes of online reading.